Non-working attempt to introduce := as an assignment operator

Rosuav opened the pull request:
NOT FOR MERGING

This PR is just to make the branch more visible to maintainers, incl @gvanrossum

Rosuav committed:
Non-working attempt to introduce := as an assignment operator

gvanrossum commented on Grammar/Grammar:
```diff
@@ -83,7 +83,7 @@ with_item: test ['as' expr]
 suite: simple_stmt | NEWLINE INDENT stmt+ DEDENT
 
-test: or_test ['if' or_test 'else' test] | lambdef
+test: [or_test ':='] or_test ['if' or_test 'else' test] | lambdef
```
Try `or_test [':= or_test]` instead.

Rosuav committed:
Make := non-breaking but ignored (it's just the expression)

Rosuav committed:
Implement some sort of assignment for := expressions.

There's still stuff missing (eg this doesn't mark a name as local),
but it's starting to look like what I want it to.

Rosuav committed:
Properly visit AssignExp nodes.

This appears to make them properly functional. I think.

Rosuav committed:
Implement early binding of comprehension variables in class scope

Rosuav committed:
Remove the special-casing of the outermost iterable in a comprehension

Rosuav committed:
Fix broken tests

Rosuav committed:
Add := to tokenizer

Rosuav committed:
Flag a name as early-bind only once

Rosuav committed:
Remove now-superfluous helper function

Rosuav committed:
Perform the assertion mentioned in the comment

Rosuav committed:
Adjust grammar to allow chaining

Rosuav committed:
Change iteration variable name to avoid conflict

Rosuav committed:
Remove test of implicit args now that there aren't any

Rosuav committed:
Regenerate autogeneratable files

Rosuav committed:
Update disassembly tests to match current code

Rosuav committed:
Remove tests of outermost-iterable precomputation.

These tests would remain valid if genexp priming is implemented.
